### PR TITLE
Add planned enhancement: Find criteria

### LINKED ISSUE #225: [PE-D][Tester D] Cannot find events with space in name specifically
If I know the event I want is two words I cannot fundEvent for a space character only.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: ji-just-ji/ped#4

### LINKED ISSUE #235: [PE-D][Tester B] Suggestions to feature find
**Current implementation of find**

If we have two members: Wen Xin and Hui Xing.

`findMember Xin` will only return member Wen Xin

**I suggests that:** the rule of finding can be less strict; 

for example, the command `findMember Xin` can return both Wen Xin and Hui Xing.

It may benefits users as users may remember members name wrongly. 

It will be useful if the find can return members whose name only contains part of the keyword.

**After adding command below**, there is `0 members listed!`:

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: KumChaiYin/ped#8

### PR DESCRIPTION
**Pull request by dylkaw**
Closes #235 and closes #225.

What have you done in this PR?
* Added planned enhancements to `findMember` and `findEvent` commands to find all matches of given keywords.

### CODE REVIEW

File: docs/DeveloperGuide.md
Comment (SelwynAng) [inline docs/DeveloperGuide.md]: Suggestion to change this to Improve `findMember` and `findEvent` criteria

File: docs/DeveloperGuide.md
Comment (SelwynAng) [inline docs/DeveloperGuide.md]: Suggestion to captialise the 's' in solution

**Review by SelwynAng — APPROVED** [review]
LGTM aside from some formatting nitpicks

File: docs/DeveloperGuide.md
Comment (dylkaw) [reply docs/DeveloperGuide.md]: Done!

File: docs/DeveloperGuide.md
Comment (dylkaw) [reply docs/DeveloperGuide.md]: Done!

**Review by wr1159 — APPROVED** [review]
LGTM now, okay to merge once merge conflict is fixed!

**Comment by dylkaw** [comment]
> LGTM now, okay to merge once merge conflict is fixed!

Merge conflict has been fixed!

File: docs/DeveloperGuide.md
Comment (andrechuakj) [inline docs/DeveloperGuide.md]: Correct me if I'm wrong, but there should be 2 spaces after the `findEvent` command right?
```suggestion
an event name consists of 2 words, they can do `findEvent  ‎`, which would list all events which have a blank space in
```

**Review by ph-nathan — APPROVED** [review]
LGTM, same comment as Andre, and while you're here can you capitalize the s in line 475 "Proposed solution" as well? :) tho its not part of ur PR

File: docs/DeveloperGuide.md
Comment (andrechuakj) [inline docs/DeveloperGuide.md]: I think we should put these in code block:
```suggestion
### Improve `findMember`/`findEvent` criteria
```

**Pull request merged**
